src/Compilers/CSharp/Test/Syntax/Parsing/ForStatementParsingTest.cs

```diff
@@ -213,12 +216,9 @@ public void TestCommaSeparators3()
     public void TestCommaSeparators4()
     {
         UsingStatement("for (int i = 0, i) ;",
-            // (1,18): error CS1002: ; expected
-            // for (int i = 0, i) ;
-            Diagnostic(ErrorCode.ERR_SemicolonExpected, ")").WithLocation(1, 18),
-            // (1,18): error CS1525: Invalid expression term ')'
+            // (1,15): error CS1002: ; expected
             // for (int i = 0, i) ;
-            Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(1, 18),
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, ",").WithLocation(1, 15),
```

Comment:
this is a better result, as this is illegal code, andth e *vast* majority of the time, this code is hte precursor to `int i = 0; i ...`  so stating that the comma should  be a semicolon is a better error.